cpu/mips32r2_common/thread_arch.c

```diff
@@ -182,11 +182,21 @@ mem_rw(const void *vaddr)
 extern int _dsp_save(struct dspctx *ctx);
 extern int _dsp_load(struct dspctx *ctx);
 #endif
+
 /*
- * The nomips16 attribute should not really be needed, it works around a toolchain
- * issue in 2016.05-03.
+ * The official mips toolchain version 2016.05-03 needs this attribute.
+ * Newer versions (>=2017.10-05) don't. Those started being based on gcc 6,
+ * thus use that to guard the attribute.
+ * 
+ * See https://github.com/RIOT-OS/RIOT/pull/11986.
  */
+#if __GNUC__ <= 4
```

Comment:
In the comment you mention GCC 6, but then the check would be `<= 5`, no?

Reply:
I think there was no Codescape gcc toolchain based on gcc 5, they directly jumped from 4 to 6. We can only guess how gcc 5 would behave...